File: webstore/webstore-java/src/main/java/com/renan/webstore/controller/admin/CategoryController.java
package com.renan.webstore.controller.admin;

import com.renan.webstore.business.admin.CategoryService;
import com.renan.webstore.controller.form.admin.CategoryForm;
import com.renan.webstore.dto.CategoryDTO;
import com.renan.webstore.dto.mapper.CategoryDTOMapper;
import jakarta.websocket.server.PathParam;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController("/api/admin/category")
@RequiredArgsConstructor
public class CategoryController {
    
    private final CategoryService service;
    private final CategoryDTOMapper mapper;
    
    @GetMapping
    public List<CategoryDTO> getCategories(){
        return service.getCategories().stream().map(mapper).toList();
    }
    
    @PostMapping
    public String createCategory(@RequestBody CategoryForm form){
        return service.createCategory(form);
    }
    
    @PutMapping("/{id}")
    public void updateCategory(@RequestBody CategoryForm form, @PathParam("id")String id) {
        service.updateCategory(form, id);
    }
    
    @DeleteMapping("/{id}")
    public void deleteCategory(@PathParam("id")String id) {
        service.deleteCategory(id);
    }
    
}

File: webstore/webstore-kotlin/src/main/kotlin/com/renan/webstore/controller/admin/CategoryController.kt
package com.renan.webstore.controller.admin

import com.renan.webstore.business.CategoryService
import com.renan.webstore.controller.admin.form.CategoryForm
import com.renan.webstore.dto.CategoryDTO
import jakarta.websocket.server.PathParam
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/admin/category")
class CategoryController(
    val categoryService: CategoryService
) {

    @GetMapping
    fun getCategories(): List<CategoryDTO> = categoryService.getCategories()

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createCategory(@RequestBody form: CategoryForm): CategoryDTO = categoryService.createCategory(form)

    @PutMapping("/{id}")
    fun updateCategory(
        @RequestBody form: CategoryForm,
        @PathVariable("id") id: Int
    ) = categoryService.updateCategory(form, id)

    @DeleteMapping(value=["/{id}"])
    fun deleteCategory(@PathVariable("id") id: Int) = categoryService.deleteCategory(id)

}